src/lib.rs

```diff
@@ -522,6 +540,15 @@ impl<S: SuffixScheme> FileRotate<S> {
 
         self.file = Some(File::create(&self.basepath)?);
 
+        if let Some(_) = &self.file {
+            let metadata = metadata(&self.basepath).unwrap();
```

Comment:
try to avoid unwrap. You can remove this line and change the next line to something like:
```
if let Ok(time) = metadata(&self.basepath).and_then(|metadta| metadata.modified()) {
```